javatests/arcs/core/storage/DirectStoreMuxerTest.kt

```diff
@@ -99,4 +159,221 @@ class DirectStoreMuxerTest {
         }
         assertThat(callbacks).isEqualTo(1)
     }
+
+    @Test
+    fun propagatesModelUpdates_fromProxyMuxer_toDrivers() = runBlockingTest {
+        val driverProvider = MockDriverProvider()
+        DriverFactory.register(driverProvider)
+
+        val directStoreMuxer = createDirectStoreMuxer()
+
+        val callbackId = directStoreMuxer.on(ProxyCallback {})
+
+        val entityCrdtA = createPersonEntityCrdt("bob", 42)
+
+        assertThat(
+            directStoreMuxer.onProxyMessage(
+                ProxyMessage.ModelUpdate(entityCrdtA.data, id = callbackId, muxId = "a")
+            )
+        ).isTrue()
+
+        val driverA = directStoreMuxer.getEntityDriver("a")
+        val capturedA = driverA.sentData.first()
+        assertThat(capturedA.toRawEntity().singletons)
+            .containsExactly(
+                "name", "bob".toReferencable(),
+                "age", 42.toReferencable()
+            )
+        assertThat(directStoreMuxer.stores.size).isEqualTo(1)
+
+        val entityCrdtB = createPersonEntityCrdt("alice", 10)
+
+        assertThat(
+            directStoreMuxer.onProxyMessage(
+                ProxyMessage.ModelUpdate(entityCrdtB.data, id = callbackId, muxId = "b")
+            )
+        ).isTrue()
+
+        val driverB = directStoreMuxer.getEntityDriver("b")
+        val capturedB = driverB.sentData.first()
+        assertThat(capturedB.toRawEntity().singletons)
+            .containsExactly(
+                "name", "alice".toReferencable(),
+                "age", 10.toReferencable()
+            )
+        assertThat(directStoreMuxer.stores.size).isEqualTo(2)
+    }
+
+    @Test
+    fun propagatesModelUpdate_fromProxyMuxer_toDriver_toOtherProxyMuxers() = runBlockingTest {
+        val driverProvider = MockDriverProvider()
+        DriverFactory.register(driverProvider)
+
+        val directStoreMuxer = createDirectStoreMuxer()
+
+        val job = Job(coroutineContext[Job])
+
+        // proxy muxer that sends a model update to direct store muxer
+        var callbackInvoked = false
+        val callbackId1 = directStoreMuxer.on(ProxyCallback {
+            callbackInvoked = true
+        })
+
+        // proxy muxer that receives a model update from direct store muxer
+        val callbackId2 = directStoreMuxer.on(ProxyCallback {
+            assertThat(it is ProxyMessage.ModelUpdate).isTrue()
+            job.complete()
+        })
+
+        // set up store for muxId "a" and register for each proxy muxer
+        val muxIdA = "a"
+        val (idSet, store) = directStoreMuxer.store(muxIdA)
+        directStoreMuxer.registerToStore(callbackId1, muxIdA, idSet, store)
+        directStoreMuxer.registerToStore(callbackId2, muxIdA, idSet, store)
```

Comment:
it would be better to test using public methods (and make registerStore private). For example, can you use getLocalData(muxIdA, callbackId1)?
actually, you shouldn't need to register callbackId1 since it will be automatically registered when you send the proxy message below (and it would be good to verify that).
So just 
getLocalData(muxIdA, callbackId2)
could work?

Reply:
Good call. Done.